### PR TITLE
Remove the service/client split and access /dev/sg* directly

### LINKED ISSUE #26: Missing validation of sg device path
While the comments in the code suggest that the device parameter should always point to a /dev/sg* there is no actual validation of the device path, so the utility will blindly run the ioctls against whatever path is given. While I have a hard time constructing a security issue out of a fixed set of ioctl calls, it certainly seems sloppy to not do any sanity checks.

I don't see any reason to allow paths outside of /dev, also validating that the given SG device is actually a usbsdmux seems appropriate as to not confuse unrelated SG devices with the usbsdmux specific ioctls if the user happens to specify the wrong device by accident or malice.

### PR DESCRIPTION
**Pull request by hnez**
This change is based on the work done by sephalon #23 , who discovered that CAP_SYS_RAWIO is not required to perform the ioctls on /dev/sg* devices when the device is opened for writing.

The original suggestion was to use this fact to run the service as a user with write access to the /dev/sg* devices belonging to USB-SD-Muxes and grant access to the service socket to users based on group membership.

We instead decided to grant direct access to the /dev/sg* devices to the end-users based on group membership and deprecate the service altogether.

This closes #26 and closes #23 due to the removal of the service, which used to be an ioctl oracle.

I suggest merging this change into the next branch, and perform extensive testing on it together with the other changes scheduled for the next release.

### CODE REVIEW

**Review by Emantor — APPROVED** [review]

File: usbsdmux/__main__.py [hunk 1]
```diff
@@ -20,43 +20,7 @@
...
-    payload["mode"] = mode
-    payload["sg"] = sg
-    sock.send(json.dumps(payload).encode())
-    answer = json.loads(sock.recv(4096).decode())
-    sock.close()
-    if 'text' in answer:
-        print(answer['text'])
-    if not 'error' in answer or answer['error']:
-        exit(1)
+import sys
```
Comment (Bastian-Krause) [inline usbsdmux/__main__.py:23]: Wrong import order.

File: usbsdmux/__main__.py [hunk 2]
```diff
@@ -74,39 +38,43 @@ def main():
...
+              "Please upgrade your scripts to not supply either of these arguments",
+              file=sys.stderr)
+
+    ctl = UsbSdMux(args.sg)
+    mode = args.mode.lower()
+
+    if mode == "off":
+        ctl.mode_disconnect()
+
+    elif mode == "dut" or mode == "client":
```
Comment (Bastian-Krause) [inline usbsdmux/__main__.py:69]:
```suggestion
    elif mode in ("dut", "client"):
```

**Review by SmithChart — APPROVED** [review]

File: usbsdmux/service.py [hunk 3]
```diff
@@ -18,152 +18,21 @@
...
-* 'error' -- Boolean: True if an error occurred. False on success.
-* 'text'  -- Text that describes the result.
-             Contains the Text of an exception if one occurred.
-
-For example:
-{"error": False, "text": "Success"}
-or as a result of mode=get:
-{"error": False, "text": "dut"}
+This file is kept here to notify users that have set up a systemd service.
 """
```
Comment (Bastian-Krause) [inline usbsdmux/service.py:32]: Docstring in wrong position.

**Comment by Bastian-Krause** [comment]
Note that my review comments were not addressed before this PR was merged into next.

**Pull request merged**